git-link.el

```diff
@@ -176,6 +188,22 @@
 	"origin"
       remote)))
 
+(defun git-link--handler (str alist n)
+  "Return Nth element from ALIST whose `car' (a regexp) matches STR.
+
+The ALIST consists of (REGEXP FN) list elements.
+Valid ALISTs are `git-link-commit-remote-alist',`git-link-commit-alist'.
+
+Nth element of ALIST is returned at the first match of STR with REGEXP.
+
+Return nil,
+- if STR does not match with REGEXP in any of the elements of ALIST, or
+- if STR is not a string "
+  (when (stringp remote-host)
+    (nth n (cl-find-if (lambda (lst)
+                         (string-match-p (car lst) remote-host))
```

Comment:
Fixed. Also enabled lexical-binding. Now compiles fine except for one free variable warning for `git-timemachine-revision`.